### PR TITLE
sort_topic_nodes added to TreeNode

### PR DESCRIPTION
**Pull request by richard-dinh**
@kollivier  Creating PR due to request: https://github.com/learningequality/sushi-chef-chicken-and-chips/pull/1#discussion_r579429868

### CODE REVIEW

File: ricecooker/classes/nodes.py [hunk 1]
```diff
@@ -427,6 +427,26 @@ def truncate_fields(self):
 
         super(TreeNode, self).truncate_fields()
 
+    def sort_topic_nodes(self, channel, key = None, reverse = False):
```
Comment (kollivier) [inline ricecooker/classes/nodes.py:430]: Please name this `sort_child_nodes` and remove the `channel` argument. We should sort `self.children`.

File: ricecooker/classes/nodes.py [hunk 2]
```diff
@@ -427,6 +427,26 @@ def truncate_fields(self):
...
+        :param reverse: A Boolean. False will sort ascending, True will sort descending. False by default
+        :return: Sorted channel
+        """
+        # default natural sorting
+        if not key:
+            convert = lambda text: int(text) if text.isdigit() else text.lower() 
+            key = lambda key: [ convert(re.sub(r'[^A-Za-z0-9]+', '', c.replace('&', 'and'))) for c in re.split('([0-9]+)', key.title) ]
+        try:
+            channel.children = sorted(channel.children, key = key, reverse = reverse)
+        except Exception as e:
```
Comment (kollivier) [inline ricecooker/classes/nodes.py:444]: Please remove the exception part. If the developer wants the default sorting, they can write fallback code themselves.

**Review by kollivier — CHANGES_REQUESTED** [review]
Thanks! Needs a couple tweaks but otherwise looks good!

File: ricecooker/classes/nodes.py (hunk 1 above)
Comment (kollivier) [reply ricecooker/classes/nodes.py:430]: Also make sure to update the function doc strings to remove references to channel.

Comment (richard-dinh) [reply ricecooker/classes/nodes.py]: changes have been made

File: ricecooker/classes/nodes.py (hunk 2 above)
Comment (richard-dinh) [reply ricecooker/classes/nodes.py]: removed exception

**Comment by kollivier** [comment]
LGTM, thanks!

**Pull request merged**